### ISSUE TITLE
Failed install in independence environment

### ISSUE THREAD

**Issue by conao3** [emacs-elsa/flycheck-elsa#9]
Hi!
Because I attempt to research this package something wrong, I want to run this package in an independent environment.

This file saved at `~/.debug.emacs.d/elsa/init.el` and load it with `emacs -q -l ~/.debug.emacs.d/elsa/init.el`
### init.el
```
;; ~/.debug.emacs.d/elsa/init.el

;; you can run like 'emacs -q -l ~/.debug.emacs.d/{{pkg}}/init.el'
(when load-file-name
  (setq user-emacs-directory
        (expand-file-name (file-name-directory load-file-name))))

(prog1 "prepare leaf"
  (prog1 "package"
    (custom-set-variables
     '(package-archives '(("org"   . "https://orgmode.org/elpa/")
                          ("melpa" . "https://melpa.org/packages/")
                          ("gnu"   . "https://elpa.gnu.org/packages/"))))
    (package-initialize))

  (prog1 "leaf"
    (unless (package-installed-p 'leaf)
      (package-refresh-contents)
      (package-install 'leaf))))



(load (locate-user-emacs-file "../essentials.el"))

(leaf flycheck
  :ensure t
  ;; :custom ((global-flycheck-mode . t))
  :config
  (leaf flycheck-elsa
    :ensure t
    :hook ((flycheck-mode-hook . flycheck-elsa-setup))
    :config
    (leaf elsa-font-lock            ; elsa built-in
      :require t
      :defun (elsa-setup-font-lock)
      :config (elsa-setup-font-lock))))
```

### install log
```
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/leaf-20191023.1053’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/leaf-20191023.1053/leaf.el at Tue Nov 12 17:43:23 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/leaf-20191023.1053/’

Compiling no file at Tue Nov 12 17:43:23 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/real-auto-save-20190224.1446’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/real-auto-save-20190224.1446/real-auto-save.el at Tue Nov 12 17:43:23 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/real-auto-save-20190224.1446/’

Compiling no file at Tue Nov 12 17:43:24 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/epl-20180205.2049’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/epl-20180205.2049/epl.el at Tue Nov 12 17:43:24 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/epl-20180205.2049/’

Compiling no file at Tue Nov 12 17:43:25 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/pkg-info-20150517.1143’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/pkg-info-20150517.1143/pkg-info.el at Tue Nov 12 17:43:25 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/pkg-info-20150517.1143/’

Compiling no file at Tue Nov 12 17:43:26 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/dash-20191109.1327’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/dash-20191109.1327/dash.el at Tue Nov 12 17:43:26 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/dash-20191109.1327/’

Compiling no file at Tue Nov 12 17:43:33 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/flycheck-20191108.2129’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/flycheck-20191108.2129/flycheck-ert.el at Tue Nov 12 17:43:33 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/flycheck-20191108.2129/’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/flycheck-20191108.2129/flycheck.el at Tue Nov 12 17:43:33 2019

Compiling no file at Tue Nov 12 17:43:35 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/s-20180406.808’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/s-20180406.808/s.el at Tue Nov 12 17:43:35 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/s-20180406.808/’

Compiling no file at Tue Nov 12 17:43:35 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/ansi-20150703.826’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/ansi-20150703.826/ansi.el at Tue Nov 12 17:43:35 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/ansi-20150703.826/’

Compiling no file at Tue Nov 12 17:43:39 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/package-build-20191010.616’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/package-build-20191010.616/package-build-badges.el at Tue Nov 12 17:43:39 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/package-build-20191010.616/’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/package-build-20191010.616/package-build.el at Tue Nov 12 17:43:39 2019

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/package-build-20191010.616/package-recipe-mode.el at Tue Nov 12 17:43:39 2019

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/package-build-20191010.616/package-recipe.el at Tue Nov 12 17:43:39 2019

Compiling no file at Tue Nov 12 17:43:39 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/shut-up-20180628.1830’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/shut-up-20180628.1830/shut-up.el at Tue Nov 12 17:43:39 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/shut-up-20180628.1830/’

In end of data:
shut-up.el:151:1:Warning: the following functions are not known to be defined:
    shut-up-write-region-original, shut-up-load-original

Compiling no file at Tue Nov 12 17:43:40 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/f-20191110.1357’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/f-20191110.1357/f.el at Tue Nov 12 17:43:40 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/f-20191110.1357/’

Compiling no file at Tue Nov 12 17:43:42 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/cask-20191004.1155’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/cask-20191004.1155/cask-bootstrap.el at Tue Nov 12 17:43:42 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/cask-20191004.1155/’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/cask-20191004.1155/cask.el at Tue Nov 12 17:43:42 2019
cask.el:48:1:Error: Cannot open load file: No such file or directory, /home/conao/.debug.emacs.d/elsa/cask-bootstrap

Compiling no file at Tue Nov 12 17:43:43 2019
Leaving directory ‘/home/conao/.debug.emacs.d/elsa/elpa/flycheck-elsa-20190211.1921’

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/flycheck-elsa-20190211.1921/Elsafile.el at Tue Nov 12 17:43:43 2019
Entering directory ‘/home/conao/.debug.emacs.d/elsa/elpa/flycheck-elsa-20190211.1921/’
Elsafile.el:2:2:Warning: reference to free variable ‘dash’
Elsafile.el:3:2:Warning: reference to free variable ‘eieio’
Elsafile.el:4:2:Warning: reference to free variable ‘cl’
Elsafile.el:5:2:Warning: reference to free variable ‘elsa’
Elsafile.el:6:2:Warning: reference to free variable ‘seq’
Elsafile.el:7:2:Warning: reference to free variable ‘flycheck’
Elsafile.el:11:2:Warning: reference to free variable ‘dead-code’
Elsafile.el:12:2:Warning: reference to free variable ‘style’
Elsafile.el:13:2:Warning: reference to free variable ‘error’
Elsafile.el:14:2:Warning: reference to free variable ‘variables’

In end of data:
Elsafile.el:17:1:Warning: the following functions are not known to be defined:
    register-extensions, register-ruleset

Compiling file /home/conao/.debug.emacs.d/elsa/elpa/flycheck-elsa-20190211.1921/flycheck-elsa.el at Tue Nov 12 17:43:43 2019
```

This log included error when byte-compile cask.el.
This is not a problem with Cask, and I think this problem should be solved by removing the dependency on Cask.
And I think it should be installed by functions like `flycheck-elsa-install-binary` like packages with other dependent binaries.

**Comment by Fuco1** [comment]
The dependency on cask is there because we use cask code to read the bundle file (`Cask`) and setup paths.

**Comment by conao3** [comment]
Adding Cask to a dependency does not actually resolve the Elsa dependency, and the user must run `cask install`.
If before installing `cask install`  and the Cask file exists and the dependency contains elsa (and this will destroy the global package-user-dir reported in #8), flycheck will try to run elsa, The following error is output:

```
Suspicious state from syntax checker emacs-lisp-elsa: Flycheck checker emacs-lisp-elsa returned non-zero exit code 1, but its output contained no errors: cask exec: error: Failed to execute elsa /tmp/flycheckoUeSjT/flycheck-elsa.el: [Errno 2] No such file or directory
Did you run cask install?

Try installing a more recent version of emacs-lisp-elsa, and please open a bug report if the issue persists in the latest release.  Thanks!
```

We should check if elsa can be run by executing a lightweight command with no side effects, such as `cask exec elsa -h`.

**Comment by Fuco1** [comment]
I'm not quite sure what you are trying to achieve.

If we want to run elsa, you put it as a project dev dependency and run `cask install`.  Then you can use it.  What is problemantic about that?

**Comment by conao3** [comment]
Using this package in my environment destroys the global `package-user-dir`.
I should run it in a clean environment to see if it is really a problem with this package.
However, I have found that this package cannot do that.

I also discovered that the problem was caused by using `cask-initialze`, so I am trying to fix it.
Changing the global `package-user-dir` changes the installation location of `M-x package-install`, which is not just a side effect that linter can cause
